libportability-gfx/src/conv.rs

```diff
@@ -430,3 +431,95 @@ pub fn map_image_acces(access: VkAccessFlags) -> image::Access {
 
     mask
 }
+
+pub fn map_subpass_contents(contents: VkSubpassContents) -> command::SubpassContents {
+    match contents {
+        VkSubpassContents::VK_SUBPASS_CONTENTS_INLINE => command::SubpassContents::Inline,
+        VkSubpassContents::VK_SUBPASS_CONTENTS_SECONDARY_COMMAND_BUFFERS =>
+            command::SubpassContents::SecondaryBuffers,
+
+        _ => panic!("Unexpected subpass contents: {:?}", contents),
+    }
+}
+
+pub fn map_polygon_mode(mode: VkPolygonMode, line_width: f32) -> pso::PolygonMode {
+    match mode {
+        VkPolygonMode::VK_POLYGON_MODE_FILL => pso::PolygonMode::Fill,
+        VkPolygonMode::VK_POLYGON_MODE_LINE => pso::PolygonMode::Line(line_width),
+        VkPolygonMode::VK_POLYGON_MODE_POINT => pso::PolygonMode::Point,
+        _ => panic!("Unexpected polygon mode: {:?}", mode),
+    }
+}
+
+pub fn map_cull_face(cull: VkCullModeFlags) -> Option<pso::CullFace> {
+    use super::VkCullModeFlagBits::*;
+
+    if cull == VK_CULL_MODE_NONE as _ { None }
+    else if cull == VK_CULL_MODE_FRONT_BIT as _ { Some(pso::CullFace::Front) }
+    else if cull == VK_CULL_MODE_BACK_BIT as _ { Some(pso::CullFace::Back) }
+    else if cull == VK_CULL_MODE_FRONT_AND_BACK as _ { unimplemented!() } // TODO: can we support it?
```

Comment:
we can, needs the cull face to become flags then

Reply:
DX12 for example doesn't support it as far as I know.

Reply:
yes, but we are Vulkan portability :)
on DX12, we'll treat it as disabled rasterizer

Reply:
hm, yes, that would emulate the behavior as we have the topological information about the primitives and can discard primitives if we face this cull mode and use triangles.
The only issue I currently see are pipeline statistic queries, as primitive discard happens in an earlier state than clipping (discard -> .. -> clipping -> .. -> culling), affecting the number of primitives reaching the clipping state in contrast to pure back&front culling.